File: frontend/src/types/exam.ts
export interface ExamQuestion {
  id: string;
  question: string;
  options: string[];
  correct_answer: number;
  explanation: string;
}

export interface ExamSubmission {
  user_id: string;
  answers: number[];
  time_taken: number;
}

export interface ExamResult {
  user_id: string;
  score: number;
  total_questions: number;
  correct_answers: number;
  classification: 'beginner' | 'intermediate' | 'advanced';
  submitted_at: string;
}

File: backend/models/exam.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class ExamQuestion(BaseModel):
    id: str
    question: str
    options: List[str]
    correct_answer: int
    explanation: str

class ExamSubmission(BaseModel):
    user_id: str
    answers: List[int]
    time_taken: int  # in seconds

class ExamResult(BaseModel):
    user_id: str
    score: float
    total_questions: int
    correct_answers: int
    classification: str
    submitted_at: datetime = datetime.now()